File: src/main/java/org/jukeboxmc/item/behavior/ItemWoodenSword.java
package org.jukeboxmc.item.behavior;

import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.entity.attribute.Attribute;
import org.jukeboxmc.entity.attribute.AttributeType;
import org.jukeboxmc.item.*;
import org.jukeboxmc.player.Player;
import org.jukeboxmc.util.Identifier;

import java.time.Duration;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ItemWoodenSword extends Item implements Durability, Burnable {

    public ItemWoodenSword( Identifier identifier ) {
        super( identifier );
    }

    public ItemWoodenSword( ItemType itemType ) {
        super( itemType );
    }

    @Override
    public void addToHand(@NotNull Player player ) {
        Attribute attribute = player.getAttribute( AttributeType.ATTACK_DAMAGE );
        attribute.setCurrentValue( 4 );
    }

    @Override
    public void removeFromHand(@NotNull Player player ) {
        Attribute attribute = player.getAttribute( AttributeType.ATTACK_DAMAGE );
        attribute.setCurrentValue( attribute.getMinValue() );
    }

    @Override
    public Duration getBurnTime() {
        return Duration.ofMillis( 200 );
    }

    @Override
    public int getMaxDurability() {
        return 59;
    }

    @Override
    public @NotNull ToolType getToolType() {
        return ToolType.SWORD;
    }

    @Override
    public @NotNull TierType getTierType() {
        return TierType.WOODEN;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/item/behavior/ItemWoodenSword.kt
package org.jukeboxmc.server.item.behavior

import org.jukeboxmc.api.item.*
import org.jukeboxmc.server.entity.Attribute
import org.jukeboxmc.server.item.JukeboxItem
import org.jukeboxmc.server.player.JukeboxPlayer
import java.time.Duration

class ItemWoodenSword(itemType: ItemType, countNetworkId: Boolean) : JukeboxItem(itemType, countNetworkId), Durability, Burnable {

    override fun addToHand(player: JukeboxPlayer) {
        player.getAttribute(Attribute.ATTACK_DAMAGE).apply { this.setCurrentValue(4F) }
    }

    override fun removeFromHand(player: JukeboxPlayer) {
        player.getAttribute(Attribute.ATTACK_DAMAGE).apply {
            this.setCurrentValue(this.getMinValue())
        }
    }

    override fun getBurnTime(): Duration {
        return Duration.ofMillis(200)
    }

    override fun getToolType(): ToolType {
        return ToolType.SWORD
    }

    override fun getTierType(): TierType {
        return TierType.WOODEN
    }

    override fun getMaxDurability(): Int {
        return 59
    }
}